### PR TITLE
CORTX-31242: Fix added for RGW user credentials exposed through consul KV

### PR DESCRIPTION
**Pull request by rohitkolapkar**
Signed-off-by: Rohit Kolapkar <rohit.j.kolapkar@seagate.com>

# Pull Request
## Problem Statement
-   [RGW user credentials exposed through consul KV](https://jts.seagate.com/browse/CORTX-31242)

## Design
-  During CSM agent init, we were decrypting all secrets and were storing in the config index (consul kv).
- Removed this logic as it is not correct coding style to do so. 
-  In this patch, we are decrypting the secrets as and when required and not storing it to any config index.

-   \[ \] Coding conventions are followed and code is consistent

-   \[ \] Unit and System Tests are added
-   \[ \] Test Cases cover Happy Path, Non-Happy Path and Scalability
-   \[x] Testing was performed with RPM

-   \[ \] Interface change (if any) are documented
-   \[ \] Side effects on other features (deployment/upgrade)
-   \[ \] Dependencies on other component(s)

-   \[x] JIRA number/GitHub Issue added to PR
-   \[x] PR is self reviewed
-   \[x] Jira and state/status is updated and JIRA is updated with PR link
-   \[x] Check if the description is clear and explained

-   \[ \] Changes done to WIKI / Confluence page / Quick Start Guide

## Dev Testing
1. Check login using cortxadmin user ( created during mini-provisioning ) - working fine. This means decryption of cortxadmin user password worked.
image

2. CSM User Management - working fine
image

3. Perform any IAM user operation - IAM user created successfully.
image

4. Consul KV 
image

### CODE REVIEW

File: csm/common/conf.py [hunk 1]
```diff
@@ -123,15 +122,22 @@ def decrypt_conf():
         if not cluster_id:
             raise ClusterIdFetchError("Failed to get cluster id.")
         for each_key in const.DECRYPTION_KEYS:
-            cipher_key = Cipher.generate_key(cluster_id,
-                                             conf_store.get(const.CSM_GLOBAL_INDEX,
-                                                            const.DECRYPTION_KEYS[each_key]))
-            encrypted_value = conf_store.get(const.CSM_GLOBAL_INDEX, each_key)
+            secret = conf_store.get(const.CSM_GLOBAL_INDEX, each_key)
```
Comment (pawankumarsrivastava) [inline csm/common/conf.py:125]: Remove decrypt_conf method as asked

File: csm/core/services/rgw/s3/utils.py [hunk 2]
```diff
@@ -37,10 +41,22 @@ def get_rgw_connection_config():
             const.CSM_GLOBAL_INDEX, const.RGW_S3_IAM_ADMIN_USER)
         rgw_connection_config.auth_user_access_key = Conf.get(
             const.CSM_GLOBAL_INDEX, const.RGW_S3_IAM_ACCESS_KEY)
-        rgw_connection_config.auth_user_secret_key = Conf.get(
-            const.CSM_GLOBAL_INDEX, const.RGW_S3_IAM_SECRET_KEY)
+        rgw_connection_config.auth_user_secret_key = \
```
Comment (pawankumarsrivastava) [inline csm/core/services/rgw/s3/utils.py:44]: Move this logic to service class

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: csm/common/conf.py (hunk 1 above)
Comment (rohitkolapkar) [reply csm/common/conf.py]: Done.

File: csm/core/services/rgw/s3/utils.py (hunk 2 above)
Comment (rohitkolapkar) [reply csm/core/services/rgw/s3/utils.py:44]: we can move this to service class at the time of implementing  secret decryption failover scenario. Let me know on this.

**Review by Pranali04796 — APPROVED** [review]

**Review by pawankumarsrivastava — APPROVED** [review]

**Pull request merged**